setupext.py

```diff
@@ -145,6 +161,47 @@ def check_for_openmp():
         return [], []
 
 
+def check_CPP14_flag(compile_flags):
+    # Create a temporary directory
+    ccompiler = new_compiler()
+    customize_compiler(ccompiler)
+
+    tmp_dir = tempfile.mkdtemp()
+    start_dir = os.path.abspath(".")
+
+    ok = True
+    try:
+        os.chdir(tmp_dir)
+
+        with open("test_cpp14.cpp", "w") as f:
+            f.write(CPPCODE)
+
+        os.mkdir("objects")
+
+        # Compile, link, and run test program
+        with stdchannel_redirected(sys.stderr, os.devnull):
+            ccompiler.compile(
+                ["test_cpp14.cpp"], output_dir="objects", extra_postargs=compile_flags
+            )
+    except CompileError:
+        ok = False
+    finally:
+        os.chdir(start_dir)
+    return ok
```

Comment:
reason: #3062
(you may wonder if the `finally` part is still executed after a return statement, and the answer is: yes 🎉 )
```suggestion
            )
           return True
    except CompileError:
        return False
    finally:
        os.chdir(start_dir)
```